extensions/amp-next-page/0.2/service.js

```diff
@@ -176,23 +194,37 @@ export class NextPageService {
         page.relativePos === PageRelativePos.CONTAINS_VIEWPORT
       ) {
         if (!page.isVisible()) {
-          page.setVisible(VisibilityState.VISIBLE);
+          page.setVisibility(VisibilityState.VISIBLE);
         }
-        // Hide the previous page
-        const prevPage = this.pages_[index + this.scrollDirection_];
-        if (
-          prevPage &&
-          prevPage.relativePos === PageRelativePos.LEAVING_VIEWPORT &&
-          prevPage.isVisible()
-        ) {
-          prevPage.setVisible(VisibilityState.HIDDEN);
+        // Hide the previous pages
+        let prevPageIndex = index + this.scrollDirection_;
+        while (prevPageIndex >= 0 && prevPageIndex < this.pages_.length) {
```

Comment:
This function looks to be getting quite long and nested. Can this part be extracted into a separate function?

Reply:
Will be part of #26032